### DISCUSSION TITLE
Advanced return types (FindManyArgs: include's)

### DISCUSSION THREAD

**Discussion by kevinvdburgt** [prisma/prisma#6501] [Q&A]
In our service class we have a few helper functions such as getting paginated Prisma resources:
```ts
async paginate(params: Prisma.LessonFindManyArgs): Promise<{ items: Lesson[]; total: number }> {
  const [items, total] = await Promise.all([
    this.prismaService.lesson.findMany(params),
    this.prismaService.lesson.count({ where: params.where }),
  ]);

  return { items, total };
}
```

Which works fine with quite simple calls, such as:
```ts
const { items, total } = await this.lessonsService.paginate({
  take: 50,
  skip: 0,
});
```

However, when we want to use the *FindManyArgs* include, the return type only returns the *Lesson* without the joined table *Instrument*. 
```ts
const { items, total } = await this.lessonsService.paginate({
  take: 50,
  skip: 0,
  include: {
    instrument: {
      select: { id: true, name: true },
    },
  },
});
```

How could we get access to the included *Instrument* with type safety, my guess is that I should modify the *paginate* function return type?

**Comment by ryands17** [comment]
@kevinvdburgt 👋 

Have a look at this comment on how you can get the return type based on what was passed: https://github.com/prisma/prisma/issues/3372#issuecomment-773195871

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
